rrd-graph.sh

```diff
@@ -108,15 +109,15 @@ outputprefix=${outputprefix##*/}   # extract filename
 numcpus=$(/sbin/sysctl -n hw.ncpu)
 # Get drive device names
 drivedevs=
-for i in $(/sbin/sysctl -n kern.disks | awk '{for (i=NF; i!=0 ; i--) if(match($i, '/da/')) print $i }' ); do
+for i in $(/sbin/sysctl -n kern.disks | grep da); do
   # Sanity check that the drive will return a temperature (we don't want to include non-SMART usb devices)
-  DevTemp=$(/usr/local/sbin/smartctl -a /dev/"${i}" | awk '/194 Temperature_Celsius/{print $0}' | awk '{print $10}');
-  if ! [[ "$DevTemp" == "" ]]; then
+  DevTemp=$(/usr/local/sbin/smartctl -a /dev/"${i}" | awk '/Temperature_C/{print $10}');
```

Comment:
I have written some code that I think addresses the issue. I let it run overnight and didn't see any problems on my end. Give it a spin and let me know if it works for you. I also took the opportunity to factor out rrd-lib.sh, which contains some redundant code as well as function definitions that would probably make maintaining this a little easer. There's probably more to do but I at least got the ball rolling for you.